ci/docker-publish.yml

```diff
@@ -49,32 +56,31 @@ jobs:
       packages: write
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Build image
-        run: docker build . --file Dockerfile --tag $IMAGE_NAME
-
-      - name: Log into registry
-        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u ${{ github.actor }} --password-stdin
-
-      - name: Push image
-        run: |
-          IMAGE_ID=docker.pkg.github.com/${{ github.repository }}/$IMAGE_NAME
-
-          # Change all uppercase to lowercase
-          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
-
-          # Strip git ref prefix from version
-          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
-
-          # Strip "v" prefix from tag name
-          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
-
-          # Use Docker `latest` tag convention
-          [ "$VERSION" == "$default-branch" ] && VERSION=latest
-
-          echo IMAGE_ID=$IMAGE_ID
-          echo VERSION=$VERSION
-
-          docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
-          docker push $IMAGE_ID:$VERSION
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log into registry ${{ env.REGISTRY }}
+        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+              type=ref,event=branch
+              type=ref,event=pr
+              type=semver,pattern={{version}}
```

Comment:
It'd be best if we could get this down to zero elements. The `semver` seems required and hopefully we can change upstream to fix that.

Reply:
@clarkbw Do we want to track an issue somewhere to add the changes to upstream?

Reply:
I've tried to slim it down to only the semver version but pushing to anything but a tag failed because the push operation needs a tag.

It failed with this message:

`Error: buildx call failed with: error: tag is needed when pushing to registry`

I feel like we need to take this upstream if we wanted to minimize the tags configuration.

Reply:
Also: When removing the `event=pr` type, it technically works but we're getting warnings as well:

<img width="543" alt="image" src="https://user-images.githubusercontent.com/118549/119736321-62ca4f00-be32-11eb-9d75-d6cbae30f889.png">

Reply:
@pfleidi It's safe to remove the `tags` input for the starter workflow. [Default entries if empty](https://github.com/docker/metadata-action#tags-input) should be enough.